src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/CaseClassificationsMapper.java

```diff
@@ -3,35 +3,86 @@
-        List<CaseClassification> caseClassifications = new ArrayList<>();
-
-        val caseClassificationType = caseDetails.getCaseCategories().stream().filter(
-            element -> element.getCategoryType().equalsIgnoreCase(CaseCategoryType.CASETYPE.getLabel())
-        ).findFirst().orElseThrow(() -> new BadRequestException(CASE_TYPE_ERROR));
-
-        val caseClassificationSubType = caseDetails.getCaseCategories().stream().filter(
-            element -> element.getCategoryType().equalsIgnoreCase(CaseCategoryType.CASESUBTYPE.getLabel())
-        ).findFirst().orElseThrow(() -> new BadRequestException(CASE_SUBTYPE_ERROR));
-
-        val caseClassificationBuilder = CaseClassification.builder()
-            .caseClassificationService(caseDetails.getHmctsServiceCode());
-        caseClassificationBuilder.caseClassificationType(caseClassificationType.getCategoryValue());
-        caseClassificationBuilder.caseClassificationSubType(caseClassificationSubType.getCategoryValue());
-        caseClassifications.add(caseClassificationBuilder.build());
-        return caseClassifications;
+
+        val caseClassificationTypes = getCaseCategories(caseDetails, CaseCategoryType.CASETYPE);
+        validateSize(caseClassificationTypes,  new BadRequestException(CASE_TYPE_ERROR));
+        validateCategoriesOfCaseType(caseClassificationTypes);
+
+        val caseTypeValues = caseClassificationTypes.stream()
+            .map(element -> element.getCategoryValue())
+            .collect(Collectors.toList());
+
+        val caseClassificationSubTypes = getCaseCategories(caseDetails, CaseCategoryType.CASESUBTYPE);
+        validateSize(caseClassificationSubTypes,  new BadRequestException(CASE_SUBTYPE_ERROR));
+
+        return caseClassificationSubTypes.stream()
+            .map(caseClassificationSubType ->
+                     buildCaseClassification(
+                         caseDetails.getHmctsServiceCode(),
+                         caseClassificationSubType,
+                         caseTypeValues
+                     )
+            ).collect(Collectors.toList());
+    }
+
+    private CaseClassification buildCaseClassification(String hmctsServiceCode, CaseCategory caseClassificationSubType,
+                                                       List<String> caseClassificationTypes) {
+
+        validateCategoryParent(caseClassificationSubType, caseClassificationTypes);
+
+        return CaseClassification.builder()
+            .caseClassificationService(hmctsServiceCode)
+            .caseClassificationType(caseClassificationSubType.getCategoryParent())
+            .caseClassificationSubType(caseClassificationSubType.getCategoryValue())
+            .build();
+
+    }
+
+    private void validateCategoryParent(CaseCategory caseCategory, List<String> caseClassificationTypes) {
+        if (!(caseCategory.getCategoryParent() != null
+            && caseClassificationTypes.contains(caseCategory.getCategoryParent()))) {
```

Comment:
This might make it a little more easier to understand quickly (and works in the same way):
```suggestion
        if (caseCategory.getCategoryParent() == null
            || !caseClassificationTypes.contains(caseCategory.getCategoryParent())) {
```